x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/middleware.ts

```diff
@@ -586,7 +588,51 @@ async function getEndpointPackageInfo(
     });
   } catch (error) {
     // Ignore Errors, since this should not hinder the user's ability to use the UI
-    // eslint-disable-next-line no-console
-    console.error(error);
+    logError(error);
   }
 }
+
+/**
+ * retrieves the Endpoint pending actions for all of the existing endpoints being displayed on the list
+ * or the details tab.
+ *
+ * @param store
+ */
+const loadEndpointsPendingActions = async ({
+  getState,
+  dispatch,
+}: EndpointPageStore): Promise<void> => {
+  const state = getState();
+  const detailsEndpoint = detailsData(state);
+  const listEndpoints = listData(state);
+  const agentsIds = new Set<string>();
+
+  // get all agent ids for the endpoints in the list
+  if (detailsEndpoint) {
+    agentsIds.add(detailsEndpoint.elastic.agent.id);
+  }
+
+  for (const endpointInfo of listEndpoints) {
+    agentsIds.add(endpointInfo.metadata.elastic.agent.id);
+  }
+
+  if (agentsIds.size === 0) {
+    return;
+  }
+
+  try {
+    const { data: pendingActions } = await fetchPendingActionsByAgentId(Array.from(agentsIds));
+    const agentIdToPendingActions: AgentIdsPendingActions = new Map();
+
+    for (const pendingAction of pendingActions) {
+      agentIdToPendingActions.set(pendingAction.agent_id, pendingAction.pending_actions);
```

Comment:
I assume you are ok keeping the last pending_actions in the pendingActions array, is it right?

Reply:
You mean in the store if the code inside this `try {}` block `throw`'s an error?

Reply:
No, if you have the same pendingAction.agent_id twice (or more), you will overwrite it in the map. Just want to know if this is ok or not.

Reply:
We should not have duplicate agent ids. The code here does dedup and I think the API also dedups, so we should not get into that state. but - if we did, the data (`pending_actions`) would still be the same, so overwriting would be ok.

Reply:
thanks for the explanation 👍